lib/fluent/command/fluentd.rb

```diff
@@ -26,205 +26,206 @@
+      cmd_opts[:log_rotate_age] = Integer(age)
     rescue TypeError, ArgumentError
       usage "log-rotate-age should be #{ROTATE_AGE.join(', ')} or a number"
     end
   end
 }
 
 op.on('--log-rotate-size BYTES', 'sets the byte size to rotate log files') {|s|
-  opts[:log_rotate_size] = s.to_i
+  cmd_opts[:log_rotate_size] = s.to_i
 }
 
 op.on('--log-event-verbose', 'enable log events during process startup/shutdown') {|b|
-  opts[:log_event_verbose] = b
+  cmd_opts[:log_event_verbose] = b
 }
 
 op.on('-i', '--inline-config CONFIG_STRING', "inline config which is appended to the config file on-the-fly") {|s|
-  opts[:inline_config] = s
+  cmd_opts[:inline_config] = s
 }
 
 op.on('--emit-error-log-interval SECONDS', "suppress interval seconds of emit error logs") {|s|
-  opts[:suppress_interval] = s.to_i
+  cmd_opts[:suppress_interval] = s.to_i
 }
 
 op.on('--suppress-repeated-stacktrace [VALUE]', "suppress repeated stacktrace", TrueClass) {|b|
   b = true if b.nil?
-  opts[:suppress_repeated_stacktrace] = b
+  cmd_opts[:suppress_repeated_stacktrace] = b
 }
 
 op.on('--without-source', "invoke a fluentd without input plugins", TrueClass) {|b|
-  opts[:without_source] = b
+  cmd_opts[:without_source] = b
 }
 
 op.on('--config-file-type VALU', 'guessing file type of fluentd configuration. yaml/yml or guess') { |s|
   if (s == 'yaml') || (s == 'yml')
-    opts[:config_file_type] = s.to_sym
+    cmd_opts[:config_file_type] = s.to_sym
   elsif (s == 'guess')
-    opts[:config_file_type] = s.to_sym
+    cmd_opts[:config_file_type] = s.to_sym
   else
     usage '--config-file-type accepts yaml/yml or guess'
   end
 }
 
 op.on('--use-v1-config', "Use v1 configuration format (default)", TrueClass) {|b|
-  opts[:use_v1_config] = b
+  cmd_opts[:use_v1_config] = b
 }
 
 op.on('--use-v0-config', "Use v0 configuration format", TrueClass) {|b|
-  opts[:use_v1_config] = !b
+  cmd_opts[:use_v1_config] = !b
 }
 
 op.on('--strict-config-value', "Parse config values strictly", TrueClass) {|b|
-  opts[:strict_config_value] = b
+  cmd_opts[:strict_config_value] = b
 }
 
 op.on('--enable-input-metrics', "Enable input plugin metrics on fluentd", TrueClass) {|b|
-  opts[:enable_input_metrics] = b
+  cmd_opts[:enable_input_metrics] = b
 }
 
 op.on('--enable-size-metrics', "Enable plugin record size metrics on fluentd", TrueClass) {|b|
-  opts[:enable_size_metrics] = b
+  cmd_opts[:enable_size_metrics] = b
 }
 
 op.on('-v', '--verbose', "increase verbose level (-v: debug, -vv: trace)", TrueClass) {|b|
   if b
-    opts[:log_level] = [opts[:log_level] - 1, Fluent::Log::LEVEL_TRACE].max
+    cmd_opts[:log_level] = [default_opts[:log_level] - 1, Fluent::Log::LEVEL_TRACE].max
   end
 }
 
 op.on('-q', '--quiet', "decrease verbose level (-q: warn, -qq: error)", TrueClass) {|b|
   if b
-    opts[:log_level] = [opts[:log_level] + 1, Fluent::Log::LEVEL_ERROR].min
+    cmd_opts[:log_level] = [default_opts[:log_level] + 1, Fluent::Log::LEVEL_ERROR].min
   end
 }
 
```

Comment:
It seems possible that I'm wrong here.
I will check this. Maybe we don't have the corresponding tests.

Reply:
Fixed.